### PR TITLE
Make sure the too-small-max-bytes no false positive

### PR DESCRIPTION
**Pull request by zmstone**

### CODE REVIEW

**Review by id — APPROVED** [review]

**Comment by id** [comment]
Bump version if you want to push to master.

**Comment by zmstone** [comment]
test failed.
will have to fix code.

**Comment by zmstone** [comment]
case is still failing. because we need kafka_protocol 0.9.1
then it should fix #180

File: Makefile [hunk 1]
```diff
@@ -1,11 +1,11 @@
 PROJECT = brod
 PROJECT_DESCRIPTION = Kafka client library in Erlang
-PROJECT_VERSION = 2.3.0
+PROJECT_VERSION = 2.2.17
```
Comment (id) [inline Makefile]: 2.3.1

File: src/brod_consumer.erl [hunk 2]
```diff
@@ -346,19 +347,15 @@ handle_fetch_response(#kpro_FetchResponse{ fetchResponseTopic_L = [TopicData]
...
+handle_message_set(#kafka_message_set{messages = {?incomplete_message, Size}},
                    #state{ max_bytes = MaxBytes} = State0) ->
-  %% max_bytes is too small to fetch ONE complete message, double it.
-  NewMaxBytes = MaxBytes * 2,
-  NewMaxBytes >= (1 bsl 31) andalso
-      erlang:error({max_bytes_overflow,
-                   <<"perhaps message corruption in broker?">>}),
-  State1 = State0#state{max_bytes = NewMaxBytes},
+  %% max_bytes is too small to fetch ONE complete message
+  State1 = State0#state{max_bytes = Size},
```
Comment (id) [inline src/brod_consumer.erl]: check that Size > MaxBytes

File: Makefile (hunk 1 above)
Comment (id) [reply Makefile]: in app.src too

File: src/brod_consumer.erl [hunk 3]
```diff
@@ -350,6 +350,7 @@ handle_fetch_response(#kpro_FetchResponse{ fetchResponseTopic_L = [TopicData]
 handle_message_set(#kafka_message_set{messages = {?incomplete_message, Size}},
                    #state{ max_bytes = MaxBytes} = State0) ->
   %% max_bytes is too small to fetch ONE complete message
+  true = Size < MaxBytes, %% assert
```
Comment (id) [inline src/brod_consumer.erl]: `>`

**Comment by id** [comment]
update kafka_protocol version in rebar.config

File: test/brod_consumer_SUITE.erl [hunk 4]
```diff
@@ -406,6 +454,30 @@ make_ts_str() ->
 make_bytes(Bytes) ->
   iolist_to_binary(lists:duplicate(Bytes, 0)).
 
+%% Wait in a loop for the max_bytes to be tried in brod_consumer
+%% The trigger of sending fetch request is non-deterministic
+%% but the retry sequence is known to be:
+%% N, N ... N*2 ... N*4
```
Comment (id) [inline test/brod_consumer_SUITE.erl]: not exactly true, but ok

**Review by id — APPROVED** [review]

**Pull request merged**